src/openapi_parser/builders/schema.py

```diff
@@ -79,8 +84,10 @@ def merge_all_of_schemas(original_data: dict) -> dict:
 
 class SchemaFactory:
     _builders: Dict[DataType, SchemaBuilderMethod]
+    strict_enum: bool
 
-    def __init__(self) -> None:
+    def __init__(self, strict_enum: bool = True) -> None:
+        self.strict_enum = strict_enum
```

Comment:
I stuck with the existing pattern, but I think this could also be a data class with `__post_init__` implemented to set up the `_builders` map.

Reply:
ideally, data classes should not contain any logic and transformation rules besides simple type-casting, for those cases using factories and builders is much better, but yeah, it could be simplified

Reply:
I wasn't aware of that convention yet, TIL!

I often use them as datawrappers/namespaces with some factory utilities, but that's just my personal approach. I'm sticking to the existing code style in your project :)